issue_metrics.py

```diff
@@ -121,7 +121,10 @@ def auth_to_github() -> github3.GitHub:
         github3.GitHub: A github api connection.
     """
     if token := os.getenv("GH_TOKEN"):
-        github_connection = github3.login(token=token)
+        if os.getenv("GITHUB_SERVER_URL") == 'https://github.com':
```

Comment:
Some users run this locally and if `GITHUB_SERVER_URL` is not set, the code exits with a failure. This suggestion maintains the current functionality for users not utilizing github actions while also allowing the pass through of the `GITHUB_SERVER_URL` when using actions.

```suggestion
        if not os.getenv("GITHUB_SERVER_URL"):
            github_connection = github3.login(token=token)
        elif os.getenv("GITHUB_SERVER_URL") == 'https://github.com':
```